Dockerfile

```diff
@@ -3,134 +3,153 @@ MAINTAINER Michael Heinold @ DKFZ
+    && chmod 777 /data/datastore /data/binaries
+
+# install perl modules
+RUN cpanm Math::CDF
+
+# install python modules
+RUN pip install --index-url https://pypi.python.org/simple/ --upgrade pip && hash -r
+RUN pip install pysam==0.8.0 \
+    && pip install numpy==1.7.0 \
+    && pip install python-dateutil \
+    && pip install matplotlib==1.0.1 \
+    && pip install Biopython==1.57 \
+    && pip install scipy==0.12.0
 
 #Grid engine setup - This is taken from the pancancer setup for SGE clusters.
-RUN export DEBIAN_FRONTEND=noninteractive; \
-	echo `head -n 1 /etc/hosts | cut -f 1` master > /tmp/hostsTmp && tail -n +2 /etc/hosts >> /tmp/hostsTmp && cp /tmp/hostsTmp /etc/hosts && echo master > /etc/hostname; \
-	hostName=`hostname`; \
-    export HOST=$hostName; \
-    apt-get -q -y --force-yes install gridengine-client gridengine-common gridengine-exec gridengine-master; \
-    /etc/init.d/gridengine-exec stop; \
-    /etc/init.d/gridengine-master restart; \
-    /etc/init.d/gridengine-exec start;
+RUN export DEBIAN_FRONTEND=noninteractive \
+    && echo `head -n 1 /etc/hosts | cut -f 1` master > /tmp/hostsTmp \
+    && tail -n +2 /etc/hosts >> /tmp/hostsTmp \
+    && cp /tmp/hostsTmp /etc/hosts \
+    && echo master > /etc/hostname \
+	&& hostName=`hostname` \
+    && export HOST=$hostName \
+    && apt-get update && apt-get install -y \
+    gridengine-client \
+    gridengine-common \
+    gridengine-exec \
+    gridengine-master \
+    && rm -rf /var/lib/apt/lists/* \
+    && /etc/init.d/gridengine-exec stop \
+    && /etc/init.d/gridengine-master restart \
+    && /etc/init.d/gridengine-exec start
 
 RUN useradd roddy -d /roddy
-    
-#RUN	
-RUN	echo `head -n 1 /etc/hosts | cut -f 1` master > /tmp/hostsTmp && tail -n +2 /etc/hosts >> /tmp/hostsTmp && cp /tmp/hostsTmp /etc/hosts && echo master > /etc/hostname; \
-	hostName=`hostname`; \
-    export HOST=$hostName; \
-    /etc/init.d/gridengine-exec stop; \
-    /etc/init.d/gridengine-master restart; \
-    /etc/init.d/gridengine-exec start; \
-    qconf -am roddy; \
-    qconf -au roddy users; \
-    qconf -as $HOST
-
-ADD scripts/sgeInit.sh /roddy/sgeInit.sh
-
-RUN echo `head -n 1 /etc/hosts | cut -f 1` master > /tmp/hostsTmp; \
-	tail -n +2 /etc/hosts >> /tmp/hostsTmp; \
-	cp /tmp/hostsTmp /etc/hosts; \
-	echo master > /etc/hostname; \
-	cat /etc/hosts; \
-	hostName=`hostname`; \
-    export HOST=$hostName; \
-    cd /roddy; chmod 777 sgeInit.sh; \
-	bash sgeInit.sh; \
-	rm sgeInit.sh;
 
-RUN mkdir /roddy/bin
-
-RUN apt-get update; easy_install Atlas; apt-get -y install libatlas-base-dev gfortran;
-
-RUN pip install --index-url=https://pypi.python.org/simple/ pip==10.0.1
-RUN pip install scipy==0.12.0
 
+RUN echo `head -n 1 /etc/hosts | cut -f1` master > /tmp/hostsTmp \
+    && tail -n +2 /etc/hosts >> /tmp/hostsTmp \
+    && cp /tmp/hostsTmp /etc/hosts \
+    && echo master > /etc/hostname \
+	&& hostName=`hostname` \
+    && export HOST=$hostName \
```

Comment:
lines 74-79 repeat lines 55-60, it's totally unnecessary.

Reply:
Hi all, 
as I worked on the initial container, I'll just give a comment, why we have these repeated hostname and gridengine steps. When we developed the container, we worked with the cache feature of Docker. When one uses this, at least in my experience, the hostname changes constantly and for SGE to work properly, the hostname MUST be set upon every RUN directive. Unfortunately I am not aware, if the current Docker files or build commands do not have this problem.